pocs/dome/astrohaven.py

```diff
@@ -148,14 +153,32 @@ def _full_move(self, send, target_feedback):
             True if the target_feedback is received from the dome before the MOVE_TIMEOUT;
             False otherwise.
         """
-        end_by = time.time() + AstrohavenDome.MOVE_TIMEOUT
-        while not self._nudge_shutter(send, target_feedback):
-            if time.time() < end_by:
-                continue
-            self.logger.error('Timed out moving the dome. Check for hardware or communications ' +
-                              'problem. send=%r latest_state=%r', send, self._read_latest_state())
-            return False
-        return True
+        # Set a short timeout on reading, so that we don't open or close slowly.
+        # In other words, we'll try to read status, but if it isn't available,
+        # we'll just send another command.
+        saved_timeout = self.serial.ser.timeout
+        self.serial.ser.timeout = 0.2
+        try:
+            end_by = time.time() + AstrohavenDome.MOVE_TIMEOUT
+            self.serial.reset_input_buffer()
+            while True:
+                self.serial.write(send)
+                data = self.serial.read_bytes(size=1)
+                if data:
+                    c = chr(data[-1])
+                    if c == target_feedback:
+                        # Woot! Moved the dome and got the desired response.
+                        return True
+                    if c != send:
+                        self.logger.debug('Unexpected value from dome: {!r}', data)
+                if time.time() < end_by:
+                    continue
+                self.logger.error(
+                    'Timed out moving the dome. Check for hardware or communications ' +
+                    'problem. send={!r} expected={!r} actual={!r}', send, target_feedback, data)
+                return False
+        finally:
```

Comment:
A `try` without an `except` but with a `finally` seems a little odd to me. I'm surprised that doesn't complain.

Reply:
finally also executes if a return is executed. See pep-341

Reply:
I forgot about that. That's potentially even more confusing. I see that this was merged already but maybe a comment explaining that.